src/ifcgeom/mapping/IfcFixedReferenceSweptAreaSolid.cpp

```diff
@@ -0,0 +1,90 @@
+/********************************************************************************
+ *                                                                              *
+ * This file is part of IfcOpenShell.                                           *
+ *                                                                              *
+ * IfcOpenShell is free software: you can redistribute it and/or modify         *
+ * it under the terms of the Lesser GNU General Public License as published by  *
+ * the Free Software Foundation, either version 3.0 of the License, or          *
+ * (at your option) any later version.                                          *
+ *                                                                              *
+ * IfcOpenShell is distributed in the hope that it will be useful,              *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 *
+ * Lesser GNU General Public License for more details.                          *
+ *                                                                              *
+ * You should have received a copy of the Lesser GNU General Public License     *
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.         *
+ *                                                                              *
+ ********************************************************************************/
+
+#include "mapping.h"
+#define mapping POSTFIX_SCHEMA(mapping)
+using namespace ifcopenshell::geometry;
+
+#ifdef SCHEMA_HAS_IfcFixedReferenceSweptAreaSolid
+
+taxonomy::ptr mapping::map_impl(const IfcSchema::IfcFixedReferenceSweptAreaSolid* inst) {
+	auto dir = map(inst->Directrix());
+	auto ref = taxonomy::cast<taxonomy::direction3>(map(inst->FixedReference()));
+	auto profile = taxonomy::cast<taxonomy::face>(map(inst->SweptArea()));
+
+	auto loft = taxonomy::make<taxonomy::loft>();
+	// @todo intialize as default
+	loft->axis = nullptr;
+
+	// @todo currently only the case is handled where directrix returns a piecewise_function
+	if (auto pwf = taxonomy::dcast<taxonomy::piecewise_function>(dir)) {
+		double start = 0;
+		double end = pwf->length();
+#ifdef SCHEMA_HAS_IfcDirectrixCurveSweptAreaSolid
+		// IfcDirectrixCurveSweptAreaSolid introduced in 4.3 changed attribute type
+		// from optional IfcParamValue to optional IfcCurveMeasureSelect.
+		// Invocation of mapping on pre-4.3 models can never result in a piecewise_function.
+		if (inst->StartParam() && inst->StartParam()->as<IfcSchema::IfcLengthMeasure>()) {
+			double s = *inst->StartParam()->as<IfcSchema::IfcLengthMeasure>();
+			if (s > start) {
+				start = s;
+			}
+		}
+		if (inst->EndParam() && inst->EndParam()->as<IfcSchema::IfcLengthMeasure>()) {
+			double e = *inst->EndParam()->as<IfcSchema::IfcLengthMeasure>();
+			if (e < end) {
+				end = e;
+			}
+		}
+#endif
+		auto len = end - start;
```

Comment:
Consider using the Settings to control the loop. You can write something like this to programmatically control the fineness of the piecewise steps

~~~
	ifcopenshell::geometry::Settings settings;
	settings.get<ifcopenshell::geometry::settings::PiecewiseStepType>().value = ifcopenshell::geometry::settings::PiecewiseStepMethod::MAXSTEPSIZE;
	settings.get<ifcopenshell::geometry::settings::PiecewiseStepParam>().value = 1.0;
~~~

These settings are then used in the following code:
https://github.com/IfcOpenShell/IfcOpenShell/blob/a60e272eee613b68dd98581dfc8654a29199bf06/src/ifcgeom/taxonomy.cpp#L456-L478

You could do something similar in this loop. Alternatively, you could just call `evaluate()` without parameters and loop over the collection of edge objects returned